src/libraries/CurrencyReserves.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+import {CustomRevert} from "./CustomRevert.sol";
+
+library CurrencyReserves {
+    using CustomRevert for bytes4;
+
+    /// @notice Thrown when a user has already synced a currency, but not yet settled
+    error AlreadySynced();
+
+    /// bytes32(uint256(keccak256("ReservesOf")) - 1)
+    bytes32 constant RESERVES_OF_SLOT = 0x1e0745a7db1623981f0b2a5d4232364c00787266eb75ad546f190e6cebe9bd95;
+    /// bytes32(uint256(keccak256("Sync")) - 1)
+    bytes32 constant CURRENCY_SLOT = 0xf0e14a408baf7f453312eec68e9b7d728ec5337fbdf671f917ee8c80f3255231;
+
+    function requireNotSynced() internal view {
+        Currency currencySynced;
+        assembly {
+            currencySynced := tload(CURRENCY_SLOT)
+        }
+        if (!currencySynced.isZero()) {
+            AlreadySynced.selector.revertWith();
+        }
+    }
+
+    function getSyncedCurrency() internal view returns (Currency currency) {
+        assembly {
+            currency := tload(CURRENCY_SLOT)
+        }
+    }
+
+    function reset() internal {
+        assembly {
+            tstore(CURRENCY_SLOT, 0)
```

Comment:
would rather this also reset the reserves of. I know its not strictly needed, but it would be weird that `getSyncedReserves` returns a non-0 value, when `getSyncedCurrency` returns 0.

Reply:
Are you fine with the extra gas costs? Will effect every non native settle...

Reply:
yes i'm ok with the extra cost

Reply:
I am happy with both... I really like the gas save, but I agree with the fact, that it will be confusing to get an outdated value with the getter function.